dts/bindings/display/sitronix,st7735r.yaml

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2020, Kim Bøndergaard <kim@fam-boendergaard.dk>
+# SPDX-License-Identifier: Apache-2.0
+
+description: ST7735R 160x128 display controller
+
+compatible: "sitronix,st7735r"
+
+include: spi-device.yaml
+
+properties:
+    reset-gpios:
+      type: phandle-array
+      required: true
+      description: RESET pin.
+
+        The RESET pin of ST7735R is active low.
+        If connected directly the MCU pin should be configured
+        as active low.
+
+    cmd-data-gpios:
+      type: phandle-array
+      required: true
+      description: D/CX pin.
+
+        The D/CX pin of ST7735R is active low (transmission command byte).
+        If connected directly the MCU pin should be configured
+        as active low.
+
+    height:
+      type: int
+      required: true
+      description: Height in pixels of the panel driven by the controller
+
+    width:
+      type: int
+      required: true
+      description: Width in pixels of the panel driven by the controller
+
+    x-offset:
+      type: int
+      required: true
+      description: The column offset in pixels of the LCD to the controller memory
+
+    y-offset:
+      type: int
+      required: true
+      description: The row offset in pixels of the LCD to the controller memory
+
+    mdac:
+      type: int
+      required: true
+      description: Memory Data Access Control
+
+    colmod:
+      type: int
+      required: true
+      description: Interface Pixel Format
+
+    pwctrl1-param:
+      type: uint8-array
+      required: true
+      description: Power Control 1 Parameter
+
+    pwctrl2-param:
+      type: uint8-array
+      required: true
+      description: Power Control 2 Parameter
+
+    pwctrl3-param:
+      type: uint8-array
+      required: true
+      description: Power Control 3 Parameter
+
+    pwctrl4-param:
+      type: uint8-array
+      required: true
+      description: Power Control 4 Parameter
+
+    pwctrl5-param:
+      type: uint8-array
+      required: true
+      description: Power Control 5 Parameter
+
+    pvgam-param:
+      type: uint8-array
+      required: true
+      description: Positive Voltage Gamma Control Parameter
+
+    nvgam-param:
+      type: uint8-array
+      required: true
+      description: Negative Voltage Gamma Control Parameter
+
+    frmctrl1-param:
+      type: uint8-array
+      required: true
+      description: Frame rate control (normal mode / full colors)
+
+    frmctrl2-param:
```

Comment:
Suggestion here: use names without `-param` suffix, and provide defaults if they can simplify usage

Reply:
Personnaly I'm also a bit concerned about the rather complex dts required.
Can default values be assigned using the UTIL_OR macro, like:
frmctrl1_param =  UTIL_OR(DT_INST_PROP(inst, frmctrl1_param, {0x01,0x2c,0x2d})

Reply:
using defaults in the driver should be ok. ILI9XXX uses DT binding file for that: https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/bindings/display/ilitek%2Cili9340.yaml#L33